### PR TITLE
improve(FillUtils): Protect getUnfilledDeposits() against RPC error

### PR DESCRIPTION
**Pull request by pxrl**
If the deposit fill status cannot be queried on the destination chain, revert to slower methods to determine the status. This fallback method is less accurate due to timing, but it allows execution to continue and the complete set of resulting deposits with unknown status will likely be reduced by subsequent filtering anyway (i.e. based on supported token, number of deposit confirmations, profitability, ...). The ultimate fill status for the residual deposits will be inferred at the time of fill simulation on the destination chain.

### CODE REVIEW

File: src/utils/FillUtils.ts [hunk 1]
```diff
@@ -97,13 +97,20 @@ export async function getUnfilledDeposits(
         // Find all unfilled deposits for the current loops originChain -> destinationChain.
         return originClient
           .getDepositsForDestinationChain(destinationChainId)
-          .filter((deposit) => deposit.blockNumber >= earliestBlockNumber);
+          .filter(({ blockNumber }) => blockNumber >= earliestBlockNumber);
```
Comment (pxrl) [inline src/utils/FillUtils.ts:100]: Driveby change - not directly related, but close enough that it made sense to modify it here.

**Comment by pxrl** [comment]
Some additional context:
Intuitively I want to place this kind of error handling directly in `fillStatusArray()`, but the decision to default to `FillStatus.Unfilled´ is opinionated. It suits the relayer where we prioritise service over absolute correctness, but not the dataworker, since this might (for example) cause it to miss legitimate slow fill requests, or to incorrectly enqueue a refund.

**Review by nicholaspai — APPROVED** [review]
